tests/js/utils.js

```diff
@@ -420,6 +421,44 @@ export const provideKeyMetrics = ( registry, extraData = {} ) => {
 	} );
 };
 
+/**
+ * Provides widget registration data to the given registry.
+ *
+ * @since n.e.x.t
+ *
+ * @param {Object}       registry     The registry to set up.
+ * @param {string}       areaSlug     Widget area slug.
+ * @param {string}       areaTitle    Widget area title.
+ * @param {string|Array} contextSlugs Widget context slug(s).
+ * @param {Object}       widgets      Object containing options mapped to widget slugs.
+ */
+export const provideWidgetRegistrations = (
+	registry,
+	areaSlug,
+	areaTitle,
+	contextSlugs = null,
+	widgets
+) => {
```

Comment:
Thank you for the kind explanation and observation.

I absolutely agree. I have created #7264, and made the current utility specific to key metrics widgets with appropriate notes. Let me know if they look good, thanks!

> Finally, just a note about the function signature that was chosen here – we generally avoid functions with more than two positional arguments because it makes it difficult to reason about what each argument is when there are many. For this reason, you'll see the `fn( a, args )` or `fn( a, options )` as conventions we use quite a bit with the difference being whether or not the second argument is required or not.

Duly noted, understood, and agreed, thank you!